### PR TITLE
New routes

### PR DESCRIPTION
**Pull request by SowmyaVijayendra**
added missing routes from prev merge

### CODE REVIEW

File: db/queries/resources.js [hunk 1]
```diff
@@ -77,11 +77,93 @@ const postResourceByUserId = (title, url, description, options, user_id) => {
...
+  return db.query(`SELECT * FROM resources WHERE id = $1;`, [resource_id])
+    .then(data => {
+      return data.rows;
+    })
+    .catch((err) => {
+      console.log(err.message);
+    });
+};
+//Add resource
+const addResource= (resource, user_id) =>{
```
Comment (TinyMito) [inline db/queries/resources.js]: Already completed line 62 for USER-STORY-01, no need for this.

File: routes/resources-api.js [hunk 2]
```diff
@@ -25,10 +25,9 @@ router.get('/', (req, res) => {
     });
 });
 
-//api to get resources only for the loggedIn user
-router.get('/users/:id', (req, res) => {
- // req.session.userId = req.params.id;
-  const userId = req.params.id;
+//api to get resources created by the loggedIn user
+router.get('/user', (req, res) => {
```
Comment (TinyMito) [inline routes/resources-api.js]: This is already completed on line 26 for USER-STORY-05, no need.

File: routes/resources-api.js [hunk 3]
```diff
@@ -43,24 +42,31 @@ router.get('/users/:id', (req, res) => {
     });
 });
 
-//api to get a specific resource for the loggedIn user
-router.get('/:id/users/:user_id', (req, res) => {
-    // req.session.userId = req.params.id;
-    console.log(req.params);
-   const userId = req.params.user_id;
+//api to get a specific resource 
+router.get('/:id', (req, res) => {
```
Comment (TinyMito) [inline routes/resources-api.js]: Please define use case for this GET request? Editing a resource for a logged in user?

If it is only getting one single resource view, this is being done in USER-STORY-04

File: routes/resources-api.js [hunk 4]
```diff
@@ -81,4 +87,83 @@ router.post('/add', (req, res) => {
     res.redirect("/my-resources");
 });
 
+//api to add a new resource
+
+router.post("/new", (req, res) => {
```
Comment (TinyMito) [inline routes/resources-api.js]: Already completed for USER-STORY-01, no need for this.

File: db/queries/resources.js [hunk 5]
```diff
@@ -77,11 +77,93 @@ const postResourceByUserId = (title, url, description, options, user_id) => {
...
+  return db.query(`INSERT INTO resources (url, title, description, user_id, topic_id)
+  VALUES ($1, $2, $3, $4, $5) RETURNING *`,
+  [resource.url, resource.title, resource.description, user_id, resource.topic_id])
+    .then(data => {      
+      return data.rows[0];
+    })
+    .catch((err) => {
+      console.log(err.message);
+    });
+};
```
Comment (IrinaGM) [inline db/queries/resources.js]: Not sure if we need it when we already have `postResourceByUserId`. Can you please remove it?

File: routes/resources-api.js [hunk 6]
```diff
@@ -25,10 +25,9 @@ router.get('/', (req, res) => {
     });
 });
 
-//api to get resources only for the loggedIn user
-router.get('/users/:id', (req, res) => {
- // req.session.userId = req.params.id;
-  const userId = req.params.id;
+//api to get resources created by the loggedIn user
```
Comment (IrinaGM) [inline routes/resources-api.js]: I am under the impression that it's covered already by `router.get('/my-resources')` in routs/view-routes.js.
If I understand correctly this is use it in the "My Resources" page.

File: routes/resources-api.js [hunk 7]
```diff
@@ -81,4 +87,83 @@ router.post('/add', (req, res) => {
...
+  newResource.user_id = userId;
+  resourceQueries
+    .addResource(newResource, userId)
+    .then((resource) => {
+      res.json({ resource });
+    })
+    .catch((err) => {
+      res.status(500).json({ error: err.message });
+    });
+});
```
Comment (IrinaGM) [inline routes/resources-api.js]: covered already by `router.get('/new-resource')` under routs/view-routes.js.

**Review by IrinaGM — CHANGES_REQUESTED** [review]
@SowmyaVijayendra please take a look at my comments, some routs are already implemented.
I think we need to finalize this and move on to the user stories we have.

File: db/queries/resources.js (hunk 5 above)
Comment (SowmyaVijayendra) [reply db/queries/resources.js]: removed

File: routes/resources-api.js (hunk 6 above)
Comment (SowmyaVijayendra) [reply routes/resources-api.js]: removed

File: routes/resources-api.js (hunk 7 above)
Comment (SowmyaVijayendra) [reply routes/resources-api.js]: removed

**Review by IrinaGM — APPROVED** [review]
approved

**Review by TinyMito — APPROVED** [review]
Looks good, please focus on user stories, we're short on time.

**Pull request merged**